ykrt/src/compile/jitc_yk/codegen/mod.rs

```diff
@@ -0,0 +1,218 @@
+//! The JIT's Code Generator.
+
+// FIXME: eventually delete.
+#![allow(dead_code)]
+
+use super::{jit_ir, CompilationError};
+use dynasmrt::ExecutableBuffer;
+#[cfg(any(debug_assertions, test))]
+use dynasmrt::{AssemblyOffset, DynasmApi};
+#[cfg(any(debug_assertions, test))]
+use std::{cell::Cell, collections::HashMap, slice};
+use typed_index_collections::TiVec;
+
+/// FIXME: one arch is hard-compiled-in, but we should be able to "cross-unit-test" different
+/// architectures.
+#[cfg(target_arch = "x86_64")]
+mod x86_64;
+
+/// Describes storage for a local variable.
+///
+/// FIXME: This assumes everything is a stack slot. When we come to implement a register allocator,
+/// this should be an enum of stack/reg.
+#[derive(Debug, Clone, Copy)]
+struct LocalAlloc {
+    /// The offset from the base pointer of the allocation.
+    ///
+    /// This is independent of which direction the stack grows. In other words, for architectures
+    /// where the stack grows downwards, you'd subtract this from the base pointer to find the
+    /// address of the allocation.
+    ///
+    /// FIXME: when we have register allocation, consider addressing stack slots by the stack
+    /// pointer, thus freeing up the base pointer for general purpose use.
+    frame_off: usize,
+}
+
+impl LocalAlloc {
+    fn new(frame_off: usize) -> Self {
+        Self { frame_off }
+    }
+}
+
+impl LocalAlloc {
+    fn frame_off(&self) -> usize {
+        self.frame_off
+    }
+}
+
+/// The result of running the code generator.
+pub(crate) struct CodeGenOutput {
```

Comment:
Wouldn't `CompiledTrace` or `ExecutableTrace` or similar be clearer?

Reply:
There's already a struct called `CompiledTrace` elsewhere, and it's not just that, it contains some stuff required for doing a disassembly with "comments" which can be matched in tests and are useful for eyeballing disassemblies.

Reply:
OK... but I think we can still find a better name than `Output`?

Reply:
Well, it's executable code plus (maybe if compiled-in) debugging/testing information.

That's quite hard to sum up in a pithy type name, hence the rather generic name.

`CodeArtifact`? `ExectableBufferWrapper`? `CodeGenFinished`?

Reply:
I think `CompiledTrace` is better than all of those, but I'm in danger of bikeshedding so I will resolve for now. It might become more obvious to us in the future.

Reply:
Problem is there's already a `CompiledTrace` in this crate, very close to this code, and that could get confusing. It's `ykrt::compile::CompiledTrace`.

Maybe they should be merged at some point.

Reply:
> Maybe they should be merged at some point.

Interesting idea. Agreed, maybe. But we can worry about that a bit later.